sale_pricelist_global_rule/models/sale_order.py

```diff
@@ -0,0 +1,94 @@
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    need_recompute_pricelist_global = fields.Boolean()
+    has_pricelist_global = fields.Boolean(compute="_compute_has_pricelist_global")
+
+    @api.depends("pricelist_id")
+    def _compute_has_pricelist_global(self):
+        for sale in self:
+            if not sale.pricelist_id:
+                sale.has_pricelist_global = False
+                continue
+            categ_ids = {}
+            prod_tmpl_ids = {}
+            for line in sale.order_line:
+                prod_tmpl_ids[line.product_id.product_tmpl_id.id] = True
+                categ = line.product_id.categ_id
+                while categ:
+                    categ_ids[categ.id] = True
+                    categ = categ.parent_id
+            categ_ids = list(categ_ids)
+            prod_tmpl_ids = list(prod_tmpl_ids)
```

Comment:
This code is repeated with pricelist one. Extract it to a method.

Reply:
done